packages/rulesets/src/spectral/functions/properties-type-object-no-definition.ts

```diff
@@ -1,23 +1,23 @@
 // If a property is defined with `'type': 'object'`, it must also define an object body.
 // RPC Code: RPC-Policy-V1-03
-const errorMessageObject =
-  "Properties with type:object that don't reference a model definition are not allowed. ARM doesn't allow generic type definitions as this leads to bad customer experience."
-const errorMessageNull =
-  "Properties with 'type' NULL are not allowed, please specify the 'type' as 'Primitive' or 'Object' referring a model."
+
+const errorMessageObject = "Properties with type:object that don't reference a model definition are not allowed. ARM doesn't allow generic type definitions as this leads to bad customer experience."
+const errorMessageNull = "Properties with 'type' NULL are not allowed, please specify the 'type' as 'Primitive' or 'Object' referring a model."
+
 export const propertiesTypeObjectNoDefinition: any = (definitionObject: any, opts: any, ctx: any) => {
   const path = ctx.path || []
 
+  // return errorMessageNull if type or properties are null
```

Comment:
Is putting the check on `definitionObject?.type` before the check on `definitionObject?.properties` necessary to resolve the false negative in your test? I'm not sure I'm following how that change resolved the false negative

Reply:
No, the reordering isn't to fix the error.

It was just to keep all the property related checks in a place.

Also, this PR adds a test that was flagged as false negative & the method seems to work fine.